### DISCUSSION TITLE
Type hint for macro

### DISCUSSION THREAD

**Discussion by MartinsRucevskis** [spatie/laravel-json-api-paginate#76] [General]
When adding this package, ide doesn't detect `jsonPaginate` macro. This can be solved by generating ide helper file, but since there are no type hints, ide doesn't have any suggestions.
There has been a pull request made before for this https://github.com/spatie/laravel-json-api-paginate/pull/57.
Could this be added?

**Comment by genesiscz** [comment]
Did you solve this somehow?

**Reply by MartinsRucevskis** [reply]
I don't think there is a solution for this. Unless type hint is added on the macro, only way around it is to add a typehint in your code or add the type hint in the generated file, which will be overwritten anyways.

**Reply by AlexVanderbist** [reply]
Feel free to open a PR adding this typehint on the macro. Thanks!
